### PR TITLE
Dockerize and Helm deploy

### PR DESCRIPTION
**Pull request by geekscruff**

### CODE REVIEW

File: chart/bin/setup.rb [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+require_relative '../../config/environment'
+
+puts 'Adding and Installing Zookeeper and ZK'
+# Note, this needs to be wrapped in Bundler.with_clean_env otherwise the zookeeper install will fail
+Bundler.with_clean_env do
+  `bundle add zookeeper`
+  `bundle add zk`
+end
+
+require 'zk'
```
Comment (geekscruff) [inline chart/bin/setup.rb]: My issue is here ... the first time this script runs, it fails

Screenshot 2020-03-23 at 15 52 03

I need to reload / refresh SOMETHING but I'm not sure what

File: chart/templates/setup-job.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        - name: setup
+          image: {{ .Values.rails.image.repository }}:{{ .Values.rails.image.tag }}
+          volumeMounts:
+            - mountPath: /home/app/webapp/public/assets
+              name: shared
+              subPath: assets
+          command: ["/bin/bash"]
+          args:
+            - "-c"
+            - "ruby ./chart/bin/setup.rb"
```
Comment (geekscruff) [inline chart/templates/setup-job.yaml]: I'm wondering if this needs to run as the app user to avoid potential permissions problems?

Comment (orangewolf) [reply chart/templates/setup-job.yaml]: yes, ideally it would. are you sure it runs as root?

File: chart/bin/setup.rb (hunk 1 above)
Comment (orangewolf) [reply chart/bin/setup.rb]: I imaging you need to reload the bundler env.  I've never added a gem to the bundle and then tried to immediatly use it.  I wonder what happens if you blow off the bundle and instead `gem install zookeeper zk`?

Comment (geekscruff) [reply chart/bin/setup.rb]: that doesn't work, the zookeeper install fails ... I went round and round this many times!

File: .env.production [hunk 3]
```diff
@@ -0,0 +1,6 @@
+REGISTRY_HOST=registry.gitlab.com
```
Comment (orangewolf) [inline .env.production]: did you mean to commit a .env.production file?

File: Gemfile [hunk 4]
```diff
@@ -65,6 +65,7 @@ gem 'arclight', github: 'sul-dlss/arclight'
 gem 'blacklight_range_limit', '~> 7.1'
 group :development, :test do
   gem 'solr_wrapper', '>= 0.3'
+  gem 'nulldb'
```
Comment (orangewolf) [inline Gemfile]: does this need to be listed seperately or does activerecord-nulldb-adapter pull it in?

File: chart/.gitignore [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# Ignore *-values files
+*-values.yml
+chart/*-values.yml
```
Comment (orangewolf) [inline chart/.gitignore]: this line (and the s below it) seem like they may not be needed

File: chart/staging-values.yaml [hunk 6]
```diff
@@ -0,0 +1,74 @@
+solr:
+  image:
+   repository: solr
+   tag: 8
```
Comment (orangewolf) [inline chart/staging-values.yaml]: looks like this file snuck in to a commit

File: config/environments/production.rb [hunk 7]
```diff
@@ -23,7 +23,7 @@
   config.public_file_server.enabled = ENV['RAILS_SERVE_STATIC_FILES'].present?
 
   # Compress JavaScripts and CSS.
-  config.assets.js_compressor = :uglifier
+  config.assets.js_compressor = Uglifier.new(harmony: true)
```
Comment (orangewolf) [inline config/environments/production.rb]: not here - but we really should figure out what in bulkrax is causing this ;-)

File: config/database.yml [hunk 8]
```diff
@@ -7,8 +7,12 @@
 default: &default
   adapter: sqlite3
```
Comment (orangewolf) [inline config/database.yml]: seems like this is going to be a problem?

File: docker-compose.production.yml [hunk 9]
```diff
@@ -0,0 +1,24 @@
+version: '2.1'
```
Comment (orangewolf) [inline docker-compose.production.yml]: this file seems unnessisary in the k8s world order ;0)

**Pull request merged**